### PR TITLE
[Enhancement] Updated Documentation for stats module

### PR DESCRIPTION
**Pull request by Sc0rpi0n101**
#### References to other Issues or PRs

Extends #16557 

#### Brief description of what is fixed or changed

* Documentation updated and improved for stats module
* Docstrings are more consistent throughout the module.

#### Other comments

* Documentation can also be worked on to make it better

#### Release Notes

* stats
  * Improved Documentation for the module

### CODE REVIEW

**Comment by certik** [comment]
This is a long PR that is mixing documentation with other code fixes. Can you please split it into just documentation fix, and another PR for just the code fixes?

**Comment by Sc0rpi0n101** [comment]
> This is a long PR that is mixing documentation with other code fixes. Can you please split it into just documentation fix, and another PR for just the code fixes?

I have split this PR into only documentation fix and have created another PR for test improvements.

The test improvements are covered in #16703

File: sympy/stats/crv.py [hunk 1]
```diff
@@ -41,6 +49,27 @@ class SingleContinuousDomain(ContinuousDomain, SingleDomain):
     Represented using a single symbol and interval.
     """
     def compute_expectation(self, expr, variables=None, **kwargs):
+        """
+        Computes the expectation in Given Domain
+
+        Parameters
+        ==========
+
+        self,
```
Comment (sylee957) [inline sympy/stats/crv.py]: `self` need not be documented.

And other should follow the format, for example
https://numpydoc.readthedocs.io/en/latest/example.html#example

```
    Parameters
    ----------
    var1 : array_like
        Array_like means all those objects -- lists, nested lists, etc. --
        that can be converted to an array.  We can also refer to
        variables like `var1`.
    var2 : int
        The type above can either refer to an actual Python type
        (e.g. ``int``), or describe the type of the variable in more
        detail, e.g. ``(N,) ndarray`` or ``array_like``.
    long_var_name : {'hi', 'ho'}, optional
        Choices in brackets, default first when optional.
```
or everything will likely be rendered as bold characters.

**Review by sylee957 — COMMENTED** [review]
I think that formatting should be corrected in many places.

You may have to render it yourself after following up the tutorial in
https://github.com/sympy/sympy/tree/master/doc
and post a screenshot here as a proof.

`sphinx` with `numpydoc` extensions are quite sensitive for rst syntax, section namings, indentation, and formatting, as much as the python language itself do.

I haven't checked the actual contents though. But it may also need some reviews from people who are familiar with this module or statistics in general.

**Comment by Sc0rpi0n101** [comment]
OK, will do the changes a soon as I can

File: sympy/stats/crv.py [hunk 2]
```diff
@@ -31,6 +31,14 @@ class ContinuousDomain(RandomDomain):
     is_Continuous = True
 
     def as_boolean(self):
+        """
+        Return as is_Boolean
+
+        Raises
+        ========
+
+        NotImplementedError
```
Comment (asmeurer) [inline sympy/stats/crv.py]: I don't think this should be mentioned. This is, as I understand it, a generic superclass method which should be overridden in subclasses. The only way someone would see this documentation is if the subclass doesn't implement the method, or if they are reading the documentation for the base class specifically. So the docstring should describe what the method should implement, and mention that it needs to be defined for subclasses.

File: sympy/stats/crv.py [hunk 3]
```diff
@@ -31,6 +31,14 @@ class ContinuousDomain(RandomDomain):
     is_Continuous = True
 
     def as_boolean(self):
+        """
+        Return as is_Boolean
```
Comment (asmeurer) [inline sympy/stats/crv.py]: This isn't meaningful

File: sympy/stats/crv.py [hunk 4]
```diff
@@ -170,15 +250,26 @@ def check(*args):
...
-        """ Inverse of the CDF
+        """
+        Inverse of the CDF
 
         Used by sample
+
+        Raises
+        ======
+
+        NotImplementedError when not inverse_cdf or len(inverse_cdf) != 1
```
Comment (asmeurer) [inline sympy/stats/crv.py]: Implementation details like this should not go in the docstring.

File: sympy/stats/crv.py [hunk 5]
```diff
@@ -170,15 +250,26 @@ def check(*args):
...
+
+        Returns
+        =======
+
+        A lambda with invesr cdf function
```
Comment (asmeurer) [inline sympy/stats/crv.py]:
```suggestion
        A lambda with inverse cdf function
```

**Comment by czgdp1807** [comment]
Closing this PR. Thanks for the contributions.

**Comment by oscarbenjamin** [comment]
Why is this closed?

**Comment by czgdp1807** [comment]
Actually, IMHO, there isn't a need of documentation in the parts of stats module where this PR suggested. The reviews were also not in favour. Therefore I thought it's better not to stall this PR as no code improvements were made. However still, if other members of the community suggest that documentation of the stats module should be improved then I will continue this PR in a new one. Let me know.

**Comment by czgdp1807** [comment]
Apologies for inconveniences.

**Pull request closed without merging**